File: 0482-license-key-formatting/0482-license-key-formatting.cpp
class Solution {
public:
    string licenseKeyFormatting(string s, int k) {
        string ans;
        for(int i = s.size() - 1; i >= 0; --i) {
            if(s[i] != '-') {
                if (ans.size()%(k+1) == k) ans += '-';
                ans += toupper(s[i]);
            }
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: 0482-license-key-formatting/0482-license-key-formatting.py
class Solution:
    def licenseKeyFormatting(self, s: str, k: int) -> str:
        sb = []
        for i in range(len(s)-1, -1, -1):
            if s[i] != '-':
                if len(sb) % (k+1) == k:
                    sb.append('-')
                sb.append(s[i])
        return ''.join(sb[::-1]).upper()